server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/plan/node/DeleteTimeSeriesNode.java

```diff
@@ -0,0 +1,115 @@
+
+  private final List<PartialPath> deletedPaths;
+  private final Map<PartialPath, PlanNode> schemaRegionSpiltMap;
+  private final Map<PartialPath, List<PlanNode>> dataRegionSplitMap;
+
+  public DeleteTimeSeriesNode(PlanNodeId id, List<PartialPath> deletedPath) {
+    super(id);
+    this.deletedPaths = deletedPath;
+    this.schemaRegionSpiltMap = new HashMap<>();
+    this.dataRegionSplitMap = new HashMap<>();
+  }
+
+  public List<PartialPath> getDeletedPaths() {
+    return deletedPaths;
+  }
+
+  public Map<PartialPath, PlanNode> getSchemaRegionSpiltMap() {
+    return schemaRegionSpiltMap;
+  }
+
+  public Map<PartialPath, List<PlanNode>> getDataRegionSplitMap() {
+    return dataRegionSplitMap;
+  }
+
+  @Override
+  protected void serializeAttributes(ByteBuffer buffer) {
+    PlanNodeType.DELETE_TIMESERIES.serialize(buffer);
+    ReadWriteIOUtils.writeStringList(
+        deletedPaths.stream().map(PartialPath::getFullPath).collect(Collectors.toList()), buffer);
+  }
+
+  public static PlanNode deserialize(ByteBuffer buffer) {
+    List<PartialPath> deletedPath = new ArrayList<>();
+    ReadWriteIOUtils.readStringList(buffer).stream()
+        .map(
+            fullPath -> {
+              try {
+                return new PartialPath(fullPath);
+              } catch (IllegalPathException e) {
+                throw new IllegalArgumentException(
+                    "Cannot deserialize DeleteTimeSeriesSchemaNode", e);
+              }
+            })
+        .forEach(deletedPath::add);
+    PlanNodeId planNodeId = PlanNodeId.deserialize(buffer);
+    return new DeleteTimeSeriesNode(planNodeId, deletedPath);
+  }
+
+  @Override
+  public TRegionReplicaSet getRegionReplicaSet() {
+    return null;
+  }
+
+  @Override
+  public List<WritePlanNode> splitByPartition(Analysis analysis) {
+    for (PartialPath deletedPath : deletedPaths) {
+      String device = deletedPath.getDevice();
+      TRegionReplicaSet schemaRegionReplicaSet =
+          analysis.getSchemaPartitionInfo().getSchemaRegionReplicaSet(device);
+      DeleteTimeSeriesSchemaNode deleteTimeSeriesSchemaNode =
+          new DeleteTimeSeriesSchemaNode(this.getPlanNodeId(), deletedPath);
+      deleteTimeSeriesSchemaNode.setRegionReplicaSet(schemaRegionReplicaSet);
+      schemaRegionSpiltMap.putIfAbsent(deletedPath, deleteTimeSeriesSchemaNode);
+      List<TRegionReplicaSet> dataRegionReplicaSets =
+          analysis.getDataPartitionInfo().getDataRegionReplicaSet(device, null);
+      DeleteTimeSeriesDataNode deleteTimeSeriesDataNode =
+          new DeleteTimeSeriesDataNode(this.getPlanNodeId(), deletedPath);
+      for (TRegionReplicaSet dataRegionReplicaSet : dataRegionReplicaSets) {
+        deleteTimeSeriesDataNode.setRegionReplicaSet(dataRegionReplicaSet);
+        dataRegionSplitMap
+            .computeIfAbsent(deletedPath, k -> new ArrayList<>())
+            .add(deleteTimeSeriesDataNode);
+      }
+    }
+    return Collections.singletonList(this);
```

Comment:
I think it's better to implement this as a logical PlanNode tree construction in logicalPlanner, rather than invade into the FragmentInstanceDispatcher.